datahub-web-react/src/app/entity/shared/tabs/Documentation/components/DiscardDescriptionModal.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from 'react';
+import { Modal, Button } from 'antd';
+import { useRouteToTab } from '../../../EntityContext';
+
+type Props = {
+    buttonProps?: Record<string, unknown>;
+    isDescriptionUpdated?: boolean;
+    urn: string;
+};
+
+export const DiscardDescriptionModal = ({ buttonProps, isDescriptionUpdated, urn }: Props) => {
```

Comment:
This modal should accept 

{
     visible,
     onDiscard,
     onCancel,
}

As the props. Nothing else. It should not decide whether to show the modal. Instead, the parent DescriptionEditor should decide whether to show the modal. The parent component should also handle removing the saved localStorage state after the changes are discarded.

Reply:
Modified the DiscardDescriptionModal.ts and committed.

import React from 'react';
import { Modal, Button } from 'antd';

type Props = {
    visible?: boolean;
    onDiscard?: () => void;
    onCancel?: () => void;
};

export const DiscardDescriptionModal = ({ visible, onDiscard, onCancel }: Props) => {
    return (
        <>
            <Modal
                title="Discard Changes"
                visible={visible}
                destroyOnClose
                onCancel={onCancel}
                footer={[
                    <Button type="text" onClick={onCancel}>
                        Cancel
                    </Button>,
                    <Button onClick={onDiscard}>Discard</Button>,
                ]}
            >
                <p>Changes will not be saved. Do you want to proceed?</p>
            </Modal>
        </>
    );
};